x-pack/plugins/ml/public/file_datavisualizer/components/edit_flyout/overrides_validation.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+const INDETERMINATE_FIELD_PLACEHOLDER = '?';
+const FRACTIONAL_SECOND_SEPARATORS = ':.,';
+
+const VALID_LETTER_GROUPS = {
+  'yyyy': true,
+  'yy': true,
+  'M': true,
+  'MM': true,
+  // The simple regex here is based on the fact that the %{MONTH} Grok pattern only matches English and German month names
+  'MMM': true,
+  'MMMM': true,
+  'd': true,
+  'dd': true,
+  // The simple regex here is based on the fact that the %{DAY} Grok pattern only matches English and German day names
+  'EEE': true,
+  'EEEE': true,
+  'H': true,
+  'HH': true,
+  'h': true,
+  'mm': true,
+  'ss': true,
+  'a': true,
+  'XX': true,
+  'XXX': true,
+  'zzz': true,
+};
+// TODO: ensure this is the best way to do this check
+function isLetter(str) {
+  return str.length === 1 && str.match(/[a-z]/i);
+}
+
+export function isTimestampFormatValid(timestampFormat) {
+  const result = { isValid: true, errorMessage: null };
+
+  if (timestampFormat.indexOf('\n') >= 0 || timestampFormat.indexOf('\r') >= 0) {
+    result.isValid = false;
+    result.errorMessage = i18n.translate('xpack.ml.fileDatavisualizer.editFlyout.overrides.customTimestampValidationErrorMessage', {
+      defaultMessage: 'Multi-line timestamp formats {timestampFormat} not supported',
+      values: {
+        timestampFormat,
+      }
+    });
+    return result;
+  }
+
+  if (timestampFormat.indexOf(INDETERMINATE_FIELD_PLACEHOLDER) >= 0) {
+    result.isValid = false;
+    result.errorMessage = i18n.translate('xpack.ml.fileDatavisualizer.editFlyout.overrides.customTimestampValidationErrorMessage', {
+      defaultMessage: 'Timestamp format {timestampFormat} not supported because it contains {fieldPlaceholder}',
```

Comment:
This might be hard to understand because `fieldPlaceholder` is a question mark and there's nothing delimiting it in the message.  So the user will see something like:

> Timestamp format yyyy-MM-dd?HH:mm not supported because it contains ?

Maybe that's OK - I'll defer to the UI experts on this - but it might be clearer as:

> Timestamp format yyyy-MM-dd?HH:mm not supported because it contains a question mark character (?)

Reply:
Done in 2adb1002d6ef80fb467803db00d1273545636951